### ISSUE TITLE
How does this work now?

### ISSUE THREAD

**Issue by robinandeer** [SciLifeLab/deployment#12]
I am (read: @ingkebil is) evaluating Ansible to get stuff automated at Clinical and thought this repo should be a good starting point. Have a few thoughts/questions from my brief overview:

1. Are the roles specific enough to be reusable? The ``nas.yml`` playbook simply runs the miniconda + "nas" role :confused: Maybe nothing overlaps between processing/nas but I still have some slight hope we might be able to collaborate more closely across SciLifeLab - would you be open to breaking some stuff up into more specific roles?

2. The ``group_vars`` content is encrypted of course but I don't seem to find any place where I can get an idea of what it controls... Is this intentional "security by obscurity" or can I ask for some hints?

**Comment by guillermo-carrasco** [comment]
HI @robinandeer ,

Absolutely, a closer collaboration is always welcome. However I am not gonna get involved in this, as it wouldn't make much sense since I am leaving soon. Maybe @senthil10 ? @Galithil ? In any case:

1. I think its quite reusable for both of us, if we agree on a common directory structure. It is basically creating a conda environment, installing `TACA` and adding cronjobs to transfer data and run `TACA`; something that you also need to do, right? 

2. There is nothing about security through obscurity here :) in the `group_vars` there is only a `username`, that honestly I don't know why it is not _only_ in `roles/nas/vars/main.yaml`, I was probably experimenting with variable files hierarchy or so, don't remember.

**Comment by johandahlberg** [comment]
A small note - something that I've noticed that is really useful is to use `defaults` instead of `vars` in the roles themselves, since this can then be overriden. It shows nicely what variables are expected and it has the lowest precedence order so it can be changed by e.g. `group_vars` later.

**Comment by robinandeer** [comment]
I totally get that @guillermo-carrasco, only when I did reach out yesterday I was told you are the only person with full insight into this repo, hehe

Thanks you for responding! I will look into how little we can modify to make it work for us. We still haven't implemented TACA for our data processing but I assume you are building it general enough so this is still possible

@johandahlberg: I remember something about that, checking it out, thanks!

**Comment by guillermo-carrasco** [comment]
> I assume you are building it general enough so this is still possible

hmm... well, you'll have to judge that, but a lot of NGI specific stuff has been added lately :sweat:

**Comment by robinandeer** [comment]
Yeah, sorry... I edited that part out, didn't mean to bash. We could've been more active in the development  for sure - we just need to get _a pipeline_ up before summer
